library/alloc/src/vec/mod.rs

```diff
@@ -1460,6 +1460,21 @@ impl<T, A: Allocator> Vec<T, A> {
     /// # }
     /// ```
     ///
+    /// It is possible to initialize the spare capacity with [`spare_capacity_mut()`]:
+    ///
+    /// ```
+    /// # use core::mem::MaybeUninit;
+    /// let mut vec: Vec<u8> = Vec::with_capacity(1024);
+    /// let remaining = vec.spare_capacity_mut();
+    /// remaining[..512].iter_mut().for_each(|x| *x = MaybeUninit::new(0));
+    /// // SAFETY:
+    /// // 1. The `new_len` is less than the `capacity`.
+    /// // 2. The elements in range `0..new_len` are initialized.
+    /// unsafe {
+    ///     vec.set_len(256);
+    /// }
```

Comment:
Minor style:

```suggestion
    /// unsafe { vec.set_len(256) };
```

Reply:
Though I'd also write is like you suggest, I've based this on the existing examples for the same method and opted for consistency.